packages/argo-checkout/src/extension-points/api/post-purchase/post-purchase.ts

```diff
@@ -127,7 +127,11 @@ interface AddShippingLineChange {
   phone?: string;
 }
 
-type Changes = (AddVariantChange | AddShippingLineChange)[];
+interface SetMetafield extends Metafield {
```

Comment:
My gut tells me 

```ts
interface SetMetafield {
  type: 'set_metafield';
  metafield: Metafield
}
```

would be better / cleaner, but I can't really say why, so either is fine

Reply:
ah and it'd also impact the backend, so I guess we're probably not that inclined to change it. :D

Reply:
Hmm, it's not how we defined `AddVariant` and `AddShippingLine` either. So yeah I'll skip this. 😅  But not because it doesn't make sense (I'm good either way), but to stay consistent with other inputs too.